src/bgfx_p.h

```diff
@@ -4103,7 +4103,7 @@ namespace bgfx
 				PredefinedUniform::Enum predefined = nameToPredefinedUniformEnum(name);
 				if (PredefinedUniform::Count == predefined && UniformType::End != UniformType::Enum(type) )
 				{
-					uniforms[sr.m_num] = createUniform(name, UniformType::Enum(type), regCount);
+					uniforms[sr.m_num] = createUniform(name, UniformType::Enum(type), num);
```

Comment:
This looks good. Just you'll need to bump `BGFX_SHADER_BIN_VERSION` to 12 inside tools/shaderc/shaderc.cpp, and then add check for backward compatibility with old shaders:

```cpp
	if (isShaderVerLess(magic, 12) )
	{
		uniforms[sr.m_num] = createUniform(name, UniformType::Enum(type), regCount);
	}
	else
	{
		uniforms[sr.m_num] = createUniform(name, UniformType::Enum(type), num);
	}
```

Reply:
Also run some examples so that they work as-is without recompilation...

Reply:
Sorry for a late comment to this PR, but this change broke us because we write the shader header manually. Is this change correct? The counterpart to this change in the D3D11/D3D12 implementation still uses `regCount`. Are these both supposed to change to use `num`? What is the difference between `num` and `regCount`?

https://github.com/bkaradzic/bgfx/blob/master/src/renderer_d3d11.cpp#L4176
https://github.com/bkaradzic/bgfx/blob/master/src/renderer_d3d11.cpp#L4193

Similar for D3D12.

It's not a big issue as we can just do the same thing as shaderc and set both numbers to the element count, but I just want to confirm this is correct.

Reply:
I think it should be `num` rather than `regCount`, because `rendererUpdateUniforms()` calculate the size of data as `g_uniformTypeSize[type]*num`.

https://github.com/bkaradzic/bgfx/blob/master/src/renderer_d3d11.cpp#L4193
https://github.com/bkaradzic/bgfx/blob/master/src/bgfx.cpp#L2502

Reply:
https://github.com/bkaradzic/bgfx/blob/master/src/renderer_d3d11.cpp#L4176
 `regCount` is correct here, becuase `setPredefined()` use `m_count` as `regCount`.
https://github.com/bkaradzic/bgfx/blob/master/src/renderer.h#L141

Reply:
@bghgary https://github.com/bkaradzic/bgfx/pull/2952 Could you try this patch ?